app/services/foreman_remote_execution/renderer_methods.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+# macros to fetch information about invoked jobs
+module ForemanRemoteExecution
+  module RendererMethods
+    extend ActiveSupport::Concern
+
+    def load_hosts_invoked_job(job_id)
```

Comment:
load_* macros typically allows searching, preloading etc and returns array. How about creating new convention on for macros searching specific instance, we'd use find_* or fetch_*? Also given we don't pass any host argument, we can comit hosts from the name I think.

Reply:
Now when I think about it more, it should probably return job invocation object and not the hash. The logic loading output should be in separate methods that would be allowed in safe mode. That way, user can access more things in the report when we allow them. Lets talk about this more, perhaps we could implement load_job_invocations macro at the end :-)